### PR TITLE
Add keras model eval support

### PR DESCRIPTION
**Pull request by mingxingtan**

### CODE REVIEW

**Comment by fsx950223** [comment]
How about wrapper it as Keras callback?

File: efficientdet/coco_metric.py [hunk 1]
```diff
@@ -96,130 +213,17 @@ def estimator_metric_fn(self, detections, groundtruth_data):
...
-      self.category_ids = list(set(self.category_ids))
-      self.dataset['categories'] = [
-          {'id': int(category_id)} for category_id in self.category_ids
-      ]
-
     with tf.name_scope('coco_metric'):
       if self.testdev_dir:
-        update_op = tf.py_func(_update_op, [detections, groundtruth_data], [])
-        metrics = tf.py_func(_evaluate, [], tf.float32)
+        update_op = tf.py_func(self.update_state,
```
Comment (fsx950223) [inline efficientdet/coco_metric.py:218]: Use tf.numpy_funciton or tf.py_function.

**Comment by fsx950223** [comment]
https://github.com/google/automl/blob/c9faec9d5ebe5678c8f9e255e2d73e36874e3974/efficientdet/keras/train_lib.py#L438
Could I calculate each level loss together by stack them to a tensor, which may decrease memory usage.

**Comment by mingxingtan** [comment]
> https://github.com/google/automl/blob/c9faec9d5ebe5678c8f9e255e2d73e36874e3974/efficientdet/keras/train_lib.py#L438
> 
> 
> Could I calculate each level loss together by stack them to a tensor, which may decrease memory usage.

Yes you can, but I guess the memory impact might be limited?

**Comment by mingxingtan** [comment]
> How about wrapper it as Keras callback?

Good idea. Could you help wrap them and then use them in train.py?

**Comment by mingxingtan** [comment]
@fsx950223 If this PR looks good to you, maybe I can merge it first, so you can wrap it as a callback later?

**Comment by fsx950223** [comment]
> > How about wrapper it as Keras callback?
> 
> Good idea. Could you help wrap them and then use them in train.py?

I try it later.

**Pull request merged**